File: MockInterview.Backend/MockInterview.BLL/Models/DTOs/ContactDto.cs
using MockInterview.BLL.Models.Common;

namespace MockInterview.BLL.Models.DTOs;

/// <summary>
/// Represents contact information
/// </summary>
public class ContactDto : IEntityDto
{
    public long Id { get; set; }
    
    /// <summary>
    /// Phone number with country code
    /// </summary>
    public long PhoneNumber { get; set; }
}

File: MockInterview.Frontend/src/types/contactDto.ts
import {IMappable} from "@/types/common";

export interface IContactDto {
    id: number;
    phoneNumber: number;
}

export class ContactDto implements IMappable<IContactDto> {
    id!: number;
    phoneNumber!: number;

    mapFrom(data: IContactDto) {
        this.id = data.id;
        this.phoneNumber = data.phoneNumber;
    }
}
